### PR TITLE
[Notifier] Add support for editing Telegram messages

### PR DESCRIPTION
**Pull request by chr-hertel**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| License       | MIT

Telegram supports updating messages which is quite a use case while working with buttons to create an interactive inline interface.

See https://core.telegram.org/bots/api#editmessagetext

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php [hunk 1]
```diff
@@ -98,7 +100,7 @@ protected function doSend(MessageInterface $message): SentMessage
         if (200 !== $statusCode) {
             $result = $response->toArray(false);
 
-            throw new TransportException('Unable to post the Telegram message: '.$result['description'].sprintf(' (code %s).', $result['error_code']), $response);
+            throw new TransportException('Unable to post the Telegram message: '.$result['description'].sprintf(' (code "%s").', $result['error_code']), $response);
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php]: Maybe use post and edit accordingly?

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php (hunk 1 above)
Comment (fabpot) [inline src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php]:
```suggestion
            throw new TransportException('Unable to post the Telegram message: '.$result['description'].sprintf(' (code %d).', $result['error_code']), $response);
```

File: src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php [hunk 2]
```diff
@@ -98,7 +100,8 @@ protected function doSend(MessageInterface $message): SentMessage
         if (200 !== $statusCode) {
             $result = $response->toArray(false);
 
-            throw new TransportException('Unable to post the Telegram message: '.$result['description'].sprintf(' (code %s).', $result['error_code']), $response);
+            $verb = isset($options['message_id']) ? 'edit' : 'post';
```
Comment (fabpot) [inline src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php]: Let's remove this temp var.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chr-hertel.

**Comment by chr-hertel** [comment]
Really fast, thanks a lot! :)

**Comment by OskarStark** [comment]
As always 🤷🏼‍♂️ 😁

**Pull request merged**